harness/determined/common/experimental/trial.py

```diff
@@ -237,6 +237,7 @@ def get_trial_checkpoints(offset: int) -> bindings.v1GetTrialCheckpointsResponse
                 sortByMetric=sort_by if isinstance(sort_by, str) else None,
                 offset=offset,
                 limit=max_results,
+                states=[bindings.checkpointv1State.COMPLETED],
```

Comment:
Passing on a thread from OOB communication:

>         Only checkpoints in a ``COMPLETED`` state with a matching ``COMPLETED`` validation are considered.

Wonder what this means?

Reply:
not quite sure, but i think maybe at some point we were only accepting sort parameters that were searcher metrics, which would imply that the checkpoint had to have a searcher validation completed. 

but since we allow sorting by various attributes (state, reported time, batch number) that aren't related to validation, i don't think this comment applies any more. 

so i think we should delete the comment. what's the motivation for this change though? i'm wondering if we should accept a `states` parameter to be passed in to `list_checkpoints`

Reply:
We can have `list_checkpoints` accept state, I like that more. Would this apply to both `harness/determined/common/experimental/trial.py:list_checkpoints` and `harness/determined/common/experimental/experiment.py:list_checkpoints` ?

Reply:
Now that Michael's refreshed my memory...
> what's the motivation for this change though?

Current behavior in the CLI's `det checkpoint download` can attempt to download a checkpoint that's been deleted, and `Checkpoint.download` is [explicitly opinionated](https://github.com/determined-ai/determined/blob/8924996c129f605c6779b3b952f689ee93b73cca/harness/determined/common/experimental/checkpoint/_checkpoint.py#L209-L217) about not downloading deleted checkpoints. From https://hpe-aiatscale.atlassian.net/browse/DET-3786, looks like maybe deleted checkpoints really are deleted? Then https://hpe-aiatscale.atlassian.net/browse/DET-9491 added "partial deletion" as a feature.

Anyway. So the bug is in the CLI. I'm also not sure changing `list_checkpoints` is the right place to fix it.

I tend to not like making our interface more complicated by adding options for things that can be done in user code unless necessary. Adding a filter like `state` is one of those things.

> So i think we should delete the comment.

I like this solution to making `list_checkpoints` match its docstring, too.

Reply:
@wes-turner has a good point that adding a parameter to `list_checkpoints` has downstream implications. like it seems weird if we allow you to fetch a DELETED checkpoint but you can't actually do anything with it in the SDK. 

i do think adding `states` is a good idea, but it should probably require more thought. so my vote is we just fix the bug for now, instead of adding a new feature.

Reply:
So we can merge this fix now, and discuss SDK design in a follow-up? @azhou-determined @wes-turner